lib/google/apis/core/base_service.rb

```diff
@@ -69,6 +69,12 @@ def each
                 break if @max && item_count > @max
                 yield item
               end
+            elsif items.kind_of?(Hash)
+              items.each do |key, val|
+                item_count = item_count + 1
+                break if @max && item_count > @max
+                yield key, val
```

Comment:
Note that this is collected as an array of 2 item arrays

ex: `[ [ "key1", "value1" ], [ "key2", "value2" ] ]`

I'm not sure what the most idiomatic behavior is, please let me know.